.github/workflows/ci.yml

```diff
@@ -0,0 +1,96 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
+  workflow_dispatch: null
+
+jobs:
+  check:
+    name: "Check the code compiles"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup Rust
+        uses: dtolnay/rust-toolchain@stable
+      - name: Check
+        run: cargo check
+
+  fmt:
+    name: "Cargo fmt"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: rustfmt
+      - name: rustfmt
+        run: cargo fmt --all --check
+
+  clippy:
+    name: "Clippy"
+    needs: [check, fmt]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        features:
+          - "" # Default features
+          - "--all-features"
+          - "--all-features --tests"
+          - "--no-default-features --features secp256k1"
+          - "--no-default-features --features secp256k1,serde"
+          - "--no-default-features --features secp256k1,serde,rand"
+          - "--no-default-features --features secp256k1,serde,rand --tests"
+          - "--no-default-features --features k256"
+          - "--no-default-features --features k256,serde"
+          - "--no-default-features --features k256,serde,rand"
+          - "--no-default-features --features k256,serde,rand --tests"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup Rust
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: clippy
+      - name: Clippy
+        run: cargo clippy ${{ matrix.features }} -- -D warnings
+
+  docs:
+    needs: [check, fmt]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup Rust
+        uses: dtolnay/rust-toolchain@stable
+      - name: cargo doc
+        env:
+          RUSTDOCFLAGS: "-D rustdoc::all -A rustdoc::private-doc-tests"
+        run: cargo doc --all-features --no-deps
+
+  test:
+    needs: [check, fmt, clippy]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust: [stable, beta]
+        features:
+          - "" # Default features
+          - "--all-features"
+          - "--no-default-features --features secp256k1,serde,rand"
+          - "--no-default-features --features k256,serde,rand"
```

Comment:
Instead of duplicating the feature-flag sets to test in the CI matrix, could you please use the makefile recipes? E.g

```yaml
    strategy:
      matrix:
        rust: [stable, beta]
        recipe:
          - "test-default"
          - "test-mixed"
          - "test-secp256k1"
          - "test-k256"

    steps:
      # ...
      - name: Run tests
        run: make ${{ matrix.recipe }}
```

This makes it easier to add new feature flags in the future.